apps/api/src/modules/job/job.repository.ts

```diff
@@ -153,6 +153,77 @@ export class JobRepository {
     return selectedFields
   }
 
+  async getJobApplied(
+    titleSearch: string,
+    statusQuery: JobStatus[],
+    applicationStatus: ApplicationStatus[],
+    userId: number,
+  ): Promise<(GetJobCardDto & { ApplicationStatus })[]> {
+    const jobs = await this.prisma.job.findMany({
+      orderBy: {
+        jobId: Prisma.SortOrder.desc,
+      },
+      include: {
+        Casting: {
+          include: {
+            User: true,
+          },
+        },
+        //only get the latest application
+        Application: {
+          where: {
+            actorId: userId,
+          },
+          orderBy: {
+            applicationId: Prisma.SortOrder.desc,
+          },
+          take: 1,
+        },
```

Comment:
You can remove this, the database will be unique now

Reply:
ok